p4-16/psa/PSA.mdk

```diff
@@ -608,3 +608,209 @@ matches the in parameter of the subsequent block.
 ```
 [INCLUDE=psa.p4:Programmable_blocks]
 ```
+
+
+# PSA Portability Requirements
+
+This section lists minimum requirements for a P4 implementation to
+be PSA compliant.
+
+A PSA compliant implementation is allowed to support P4 programs that
+exceed one or more of these limits.  It is also allowed to reject such
+a P4 program as 'too complex'.
+
+A PSA compliant implementation must not reject a P4 program that obeys
+all of these limitations.  Thus P4 programs that obey all of these
+limitations should be portable across all PSA compliant
+implementations.
+
+Minimum requirements for packet parsing:
+
+- 32 bits in a parser `select` expression.
+- 50 parser states.
+- 20 parser state transitions per packet.
+- 128 bytes extracted per packet.
+- 16 bytes of metadata assigned in the parser.
```

Comment:
And in case it isn't clear from the draft text, I am expecting _many_ of these proposed numbers to change based upon review comments.  Most of the initial values I tossed in there are intended as starting points for discussion.